app/Commands/NewCommand.php

```diff
@@ -0,0 +1,511 @@
+            $this->info("You passed in '--createdb '{$dbtype}' but I do not understand");
+            $type = $this->anticipate('What type of database would you like to install? Options are '.implode(' or ', $this->dbtypes) . '.', $this->dbtypes, $this->dbtypes[0]);
+        } else {
+            $type = $dbtype;
+        }
+
+        if (! in_array($type, $this->dbtypes)) {
+            $this->alert("Now you're being silly. Entering '$type', really? Okay, I won't create a database for you.");
+        } else {
+            $this->info("I am creating a new $type database");
+            if ($this->createDatabase($type)) {
+                $this->info("I am executing 'php artisan migrate:fresh'");
+                $this->migrateFresh();
+            }
+        }
+
+    }
+
+    protected function doAuth()
+    {
+        $command = 'php artisan make:auth';
+
+        $this->info("Executing $command");
+
+        $process = new Process($command);
+        $process->setWorkingDirectory($this->projectpath);
+
+        $process->run();
+
+        if (! $process->isSuccessful()) {
+            throw new ProcessFailedException($process);
+        }
+
+        $this->line($process->getOutput());
+    }
+
+    protected function doValetLink()
+    {
+        if (! $this->hasTool('valet')) {
+            $this->warn('Cannot find valet on your system so a valet link was not created.');
+
+            return false;
+        }
+
+        $command = "valet link {$this->projectname}";
+        $this->info("Linking valet by executing '$command' in {$this->basepath}");
+
+        $process = new Process($command);
+        $process->setWorkingDirectory($this->projectpath);
+
+        $process->run();
+
+        if (! $process->isSuccessful()) {
+            throw new ProcessFailedException($process);
+        }
+
+        $this->line($process->getOutput());
+        return true;
+    }
+
+    protected function doGit()
+    {
+        if ($this->option('message')) {
+            $this->commitmessage = $this->option('message');
+        }
+
+        if (! $this->hasTool('git')) {
+            $this->info("Unable to find 'git' on the system so I cannot initialize a git repo in '{$this->projectpath}'");
+
+            return false;
+        }
+
+        $process = new Process('dummy command');
+        $process->setWorkingDirectory($this->projectpath);
+
+        $commands = [
+            'git init',
+            'git add .',
+            'git commit -m "'.str_replace('"', '\"', $this->commitmessage).'"',
+        ];
+
+        foreach ($commands as $command) {
+            $process->setCommandLine($command);
+            $process->run();
+            $this->line($process->getOutput());
+        }
+
+        return true;
+    }
+
+    protected function openTextEditor()
+    {
+        if ($this->option('editor')) {
+            $editor = $this->option('editor');
+        } else {
+            $finder = new ExecutableFinder;
```

Comment:
Feels like this `else` block can be extracted to a method named something like `getDefaultEditor`